mkosi/__init__.py

```diff
@@ -3585,26 +3607,37 @@ def make_minimal_btrfs(args: CommandLineArguments, root: str, for_cache: bool) -
         )
         f.truncate(args.root_size)
 
-        command = ["mkfs.btrfs", "-L", "root", "-d", "single", "-m", "single", "--shrink", "--rootdir", root, f.name]
-        try:
-            run(command)
-        except subprocess.CalledProcessError:
-            # The --shrink option was added in btrfs-tools 4.14.1, before that it was the default behaviour.
-            # If the above fails, let's see if things work if we drop it
-            command.remove("--shrink")
+        command = ["mkfs.btrfs", "-L", label, "-d", "single", "-m", "single", "--rootdir", root, f.name]
+
+        if args.minimal:
+            command.append("--shrink")
+            try:
+                run(command)
```

Comment:
This seems like it would clutter mkosi's output when --shrink is not supported. Would it be OK if we just grepped the --help output of mkfs.btrfs for --shrink and use that to determine whether we use it or not?

Reply:
I'm not sure that's worth it here, even on Debian stretch btrfs-tools 4.20 is available via backports. The only distro for which this might be a problem (I haven't checked) could be CentOS 7, so this should be a rather rare occurrence.

Reply:
I remember I added this back because it was missing on the CI. 

If the CI is updated we could certainly drop the compat here, but maybe we should do this independently of this, this --shrink stuff is pre-existing code after all